packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -445,20 +454,56 @@ const DataTable = ({
     ],
   );
 
-  const dataTableFilters = useMemo(
-    () =>
-      columns
-        .filter((column) => column.enableColumnFilter !== false)
-        .map((column) => {
-          return {
-            id: column.accessorKey as string,
-            label: column.header,
-            variant: column.filterVariant ?? "text",
-            options: column.filterSelectOptions,
-          } as DataFilter;
-        }),
-    [columns],
-  );
+  const convertColumnToFilter = (column: DataTableColumn<DataTableRowData>) => {
+    if (column.enableColumnFilter === false) {
+      return;
+    }
+
+    return {
+      id: column.accessorKey as string,
+      label: column.header,
+      variant: column.filterVariant ?? "text",
```

Comment:
What's this do? Default to `text`? Do we want to just force people to always pass this in, so it's clear by skimming their code what's going on?

Reply:
Also, forcing `variant` to be specified would technically be a breaking change for anyone currently consuming `DataTable` 😬